tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,255 @@
+    assert_script_run("losetup /dev/loop0");
+    assert_script_run("losetup -l -a");
+}
+
+sub run_virt_tests {
+    # Run virt-tests command for each go test files
+    my $cfg = new Config::Simple(data_url("virt_autotest/virt-tests.conf"));
+    my $section;
+    my $key;
+    my ($go_test, $skip_test, $extra_opt);
+    my ($artifacts, $junit_xml, $test_log);
+
+    foreach my $blocks (@kubevirt_tests) {
+        $section = $cfg->get_block($blocks);
+        foreach $key (keys %{$section}) {
+            # print "$key => ${$section}{$key}\n";
+            if ($key == 'GINKGO_FOCUS') {
+                $go_test = ${$section}{$key};
+            } elsif ($key == 'GINKGO_SKIP') {
+                $skip_test = "-ginkgo.skip=\"${$section}{$key}\"";
+            } elsif ($key == 'EXTRA_OPT') {
+                $extra_opt = ${$section}{$key};
+            }
+        }
+        $artifacts = "/tmp/artifacts/$section";
+        $junit_xml = "/tmp/artifacts/$section\_test.xml";
+        $test_log = "/tmp/artifacts/$section\_test.log";
+        script_run("virt-tests -ginkgo.focus=\"$go_test\" \
+                    $skip_test \
+                    -ginkgo.slowSpecThreshold 60 \
+                    -kubeconfig=/root/.kube/config \
+                    -kubectl-path=`which kubectl` \
+                    -virtctl-path=`which virtctl` \
+                    -installed-namespace=kubevirt \
+                    -deploy-testing-infra=false \
+                    -config=/usr/share/kube-virt/manifests/testing/default-config.json \
+                    -ginkgo.v=true \
+                    -test.v=true \
+                    -ginkgo.trace=true \
+                    -ginkgo.noisySkippings=false \
+                    -ginkgo.progress=true \
+                    -ginkgo.noColor \
+                    $extra_opt \
+                    -artifacts=$artifacts \
+                    -junit-output=$junit_xml \
+                    -test.testlogfile=$test_log");
+		
+        assert_script_run("sed -ie 's/Tests Suite/$section/g' $junit_xml");
+    }
+}
+
+sub generate_test_report {
+    my $result_path = "/tmp/artifacts";
+    my $build_xml = "./buildTestReports.xml";
+
+    script_run("cat > $build_xml <<__END
+<project name=\"genTestReport\" default=\"gen\" basedir=\".\">
+    <description>
+        Generate the HTML report from JUnit XML files
+    </description>
+    <target name=\"gen\">
+        <property name=\"genReportDir\" location=\"$result_path\"/>
+        <delete dir=\"$result_path/html\"/>
+        <mkdir dir=\"$result_path/html\"/>
+        <junitreport todir=\"$result_path\">
+            <fileset dir=\"$result_path\">
+                <include name=\"*_test.xml\" />
+            </fileset>
+            <report format=\"frames\" todir=\"$result_path/html\" />
+        </junitreport>
+    </target>
+</project>
+__END");
+
+    # Generate JUnit HTML aggregate test reports by ant
+    script_run("ant -buildfile $build_xml");
```

Comment:
Can the html report link be explicitly given by record_info? 

Btw, in this PR, overall, there is no enough debug info in openqa job webui to let job reviewers quickly know the key test information. Can you add more such record_info to indicate key test step start/end, and other key informations like tested kubevirt version, rke2 version, test result, etc?

Reply:
The html report is finally uploaded to <Logs & Assets> in job UI. So once the job done, we can open the link from the job.

Reply:
It seems the link is not publically accessible and we have to download the log and access 'local' xml in browser? If so, will it be unconvenient for test reviewer? Can we think about enhancing it in future PR, eg upload log to a public web server and give the final report link in job's record_info, so that reviewer can simply click it to view the final report?

Reply:
Not very sure if it can be accessible by clicking the html link after uploading. If not, we can consider to setup a web server and give a report link within the job's record_info. Will enhance it in future PR.